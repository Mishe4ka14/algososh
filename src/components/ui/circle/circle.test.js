import { Circle } from "./circle";
import React from "react";
import renderer from 'react-test-renderer';
import { ElementStates } from "../../../types/element-states";

it('circle без буквы рендерится без ошибок', () => {
    const circleWithoutLetter = renderer.create(<Circle />)
    expect(circleWithoutLetter).toMatchSnapshot();
}); 

it('circle с буквой рендерится без ошибок', () => {
    const circleWithoutLetter = renderer.create(<Circle letter="T"/>)
    expect(circleWithoutLetter).toMatchSnapshot();
}); 

it('circle с head рендерится без ошибок', () => {
    const circleWithoutLetter = renderer.create(<Circle head='head' />)
    expect(circleWithoutLetter).toMatchSnapshot();
}); 

it('circle с react-элементом в head рендерится без ошибок', () => {
    const circleWithoutLetter = renderer.create(<Circle head={<Circle />} />)
    expect(circleWithoutLetter).toMatchSnapshot();
}); 

it('circle с tail рендерится без ошибок', () => {
    const circleWithoutLetter = renderer.create(<Circle tail='tail' />)
    expect(circleWithoutLetter).toMatchSnapshot();
}); 

it('circle с реакт-элементом в tail рендерится без ошибок', () => {
    const circleWithoutLetter = renderer.create(<Circle tail={<Circle />} />)
    expect(circleWithoutLetter).toMatchSnapshot();
}); 

it('circle с index рендерится без ошибок', () => {
    const circleWithoutLetter = renderer.create(<Circle index={1} />)
    expect(circleWithoutLetter).toMatchSnapshot();
}); 

it('circle с пропом isSmall ===  true рендерится без ошибок', () => {
    const circleWithoutLetter = renderer.create(<Circle isSmall={true} />)
    expect(circleWithoutLetter).toMatchSnapshot();
}); 

it('circle в состоянии default рендерится без ошибок', () => {
    const circleWithoutLetter = renderer.create(<Circle state={ElementStates.Default} />)
    expect(circleWithoutLetter).toMatchSnapshot();
}); 

it('circle в состоянии changing рендерится без ошибок', () => {
    const circleWithoutLetter = renderer.create(<Circle state={ElementStates.Changing} />)
    expect(circleWithoutLetter).toMatchSnapshot();
}); 

it('circle в состоянии modified рендерится без ошибок', () => {
    const circleWithoutLetter = renderer.create(<Circle state={ElementStates.Modified} />)
    expect(circleWithoutLetter).toMatchSnapshot();
}); 