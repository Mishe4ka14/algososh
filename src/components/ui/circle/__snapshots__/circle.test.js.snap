// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`circle без буквы рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle в состоянии changing рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle в состоянии default рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle в состоянии modified рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle с head рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle с index рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`circle с react-элементом в head рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle с tail рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`circle с буквой рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      T
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle с пропом isSmall ===  true рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle с реакт-элементом в tail рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;
